### DISCUSSION TITLE
Modal Overlay style with compound component

### DISCUSSION THREAD

**Discussion by FelixBlaisThon** [mantinedev/mantine#4254] [Q&A]
Hello!

I've been using the Modal a lot lately and I've try to play with and without the compound component version, i.e.: 

Without compound component:
```
<Modal opened={isOpen} title="This is a title" onClose={()=>setOpen(false)}>
    {children stuff here to handle the content and footer}
</Modal>
```

With compound component: 
```
<Modal.Root opened={isOpen} onClose={()=>setOpened(false)}>
   <Modal.Overlay/>
   <Modal.Content>
      <Modal.Header>
         <Modal.Title>{This is a title}</Modal.Title>
      </Modal.Title>
   </Modal.Content>
</Modal.Root>
```

My problem is regarding the defaultProps of the Overlay component. When I use the theme to make custom style on the modal component, the `defaultProps` of the `overlayProps` object is only working for the example not using the compound component. The one using the Modal with the compound component way does not have the props applied to it.

```
ThemeOverride = {
...
   components: {
      Modal: {
         defaultProps: {
            overlayProps: {
               color: 'blue',
               opacity: 0.7
            },
         },
      },
   },
};
```

I found a workaround by doing the same theme logic, but with the `Overlay` component instead of going with the `Modal`;
```
ThemeOverride = {
...
   components: {
      Overlay: {
         defaultProps: {
            overlayProps: {
               color: 'blue',
               opacity: 0.7
            },
         },
      },
   },
};
```
 but I think it should be working with the `Modal` component.

Thanks for the help!

**Comment by rtivital** [comment]
Try defining defaultProps for ModalOverlay component instead.
